data/registry_model/quota.py

```diff
@@ -0,0 +1,415 @@
+    namespace_id: int, manifest_id: int, namespace_total: int, operation: str
+):
+    namespace_size = get_namespace_size(namespace_id)
+    namespace_size_exists = namespace_size is not None
+
+    # If backfill hasn't ran yet for this namespace don't do anything
+    if namespace_size_exists and not namespace_size.backfill_complete:
+        return
+
+    # If the namespacesize entry doesn't exist and this is the only manifest in the namespace
+    # we can assume this is the first push to the namespace and there is no blobs to be
+    # backfilled, so let the entry be created. Otherwise it still needs to be handled by the
+    # backfill worker so let's exit
+    params = {}
+    if (
+        operation == "add"
+        and not namespace_size_exists
+        and only_manifest_in_namespace(namespace_id, manifest_id)
+    ):
+        params["backfill_start_ms"] = 0
+        params["backfill_complete"] = True
+    elif operation == "add" and not namespace_size_exists:
+        return
+
+    increment_namespacesize(namespace_id, namespace_total, operation, namespace_size_exists, params)
+
+
+def write_repository_total(
+    repository_id: int, manifest_id: int, repository_total: int, operation: str
+):
+    repository_size = get_repository_size(repository_id)
+    repository_size_exists = repository_size is not None
+
+    # If backfill hasn't ran yet for this repository don't do anything
+    if repository_size_exists and not repository_size.backfill_complete:
+        return
+
+    # If the repositorysize entry doesn't exist and this is the only manifest in the repository
+    # we can assume this is the first push to the repository and there is no blobs to be
+    # backfilled, so let the entry be created. Otherwise it still needs to be handled by the
+    # backfill worker so let's exit
+    params = {}
+    if (
+        operation == "add"
+        and not repository_size_exists
+        and only_manifest_in_repository(repository_id, manifest_id)
+    ):
+        params["backfill_start_ms"] = 0
+        params["backfill_complete"] = True
+    elif operation == "add" and not repository_size_exists:
+        return
+
+    increment_repositorysize(
+        repository_id, repository_total, operation, repository_size_exists, params
+    )
+
+
+def get_namespace_id_from_repository(repository: int):
+    try:
+        repo = Repository.select(Repository.namespace_user).where(Repository.id == repository).get()
+        return repo.namespace_user_id
+    except Repository.DoesNotExist:
+        return None
+
+
+def get_namespace_size(namespace_id: int):
+    try:
+        namespace_size = (
+            QuotaNamespaceSize.select()
+            .where(QuotaNamespaceSize.namespace_user == namespace_id)
+            .get()
+        )
+        return namespace_size
+    except QuotaNamespaceSize.DoesNotExist:
+        return None
+
+
+def increment_namespacesize(
+    namespace_id: int, size: int, operation: str, exists: bool, params=None
```

Comment:
nit: increment usually implies something increasing, so I would split/factor this out into separate functions for add and subtracting. 
 
Also, can the `exists` parameter be removed, and just add the logic to this function, as opposed to have the caller responsible for figuring that out?

Reply:
I ended up moving the code in these methods to the `write_repository_total` and `write_namespace_total` methods. It makes things easier to read while fixing a bug where it would write out the total on subtraction even though the backfill wasn't completed